packages/xo-web/src/xo-app/new-vm/index.js

```diff
@@ -548,10 +548,14 @@ export default class NewVm extends BaseComponent {
     const defaultNetworkIds = this._getDefaultNetworkIds(template)
     forEach(template.VIFs, vifId => {
       const vif = getObject(storeState, vifId, resourceSet)
-      VIFs.push({
+      const networkObject = {
         network: pool || isInResourceSet(vif.$network) ? vif.$network : defaultNetworkIds[0],
-      })
+        deviceIndex: vif.device
+      }
+      VIFs.push(networkObject)
```

Comment:
Avoid introducing unnecessary changes:

```suggestion
      VIFs.push({
        device: vif.device,
        network: pool || isInResourceSet(vif.$network) ? vif.$network : defaultNetworkIds[0],
      })
```